src/Operation/All.php

```diff
@@ -23,16 +23,22 @@ final class All extends AbstractOperation
     /**
      * @pure
      *
-     * @return Closure(Iterator<TKey, T>): array<TKey, T>
+     * @return Closure(bool): Closure(Iterator<TKey, T>): list<T>|array<TKey, T>
```

Comment:
in this file we cannot use the conditional return type from the interface because we cannot reference the boolean parameter in the same way in many of the closures. So this union return type is the next best thing

Reply:
Fair enough.